src/devices/tuya.ts

```diff
@@ -2586,6 +2586,36 @@ const definitions: DefinitionWithExtend[] = [
             e.binary('motor_reversal', ea.ALL, 'ON', 'OFF'),
         ],
     },
+    {
+        fingerprint: [{modelID: 'TS130F', manufacturerName: '_TZ3000_cet6ch1r'}],
+        model: 'SC400ZB-EU',
+        vendor: 'LoraTap',
+        description: 'Curtain/blind switch',
+        fromZigbee: [fz.cover_position_tilt, fz.tuya_cover_options, tuya.fz.backlight_mode_off_on],
+        toZigbee: [
+            tz.cover_state,
+            tz.cover_position_tilt,
+            tz.tuya_cover_calibration,
+            tz.tuya_cover_reversal,
+            tuya.tz.backlight_indicator_mode_2,
+            tuya.tz.backlight_indicator_mode_1,
+        ],
+        meta: {coverInverted: true, multiEndpoint: true},
+        endpoint: (device) => {
+            return {up: 1, down: 2};
+        },
+        exposes: [
+            e.cover_position().withEndpoint('up'),
```

Comment:
It's not nice that also these exposes are under an endpoint. I propose to modify `tz.tuya_cover_calibration` instead to support up and down calibration. (e.g. `calibration_up`, `calibration_down`). Then we can also keep this in the original `TS130F` definition (add `calibration_up` and `calibration_down` exposes conditionally in the exposes function)

Reply:
I have implemented the feature following your suggestion.
I also update `fz.tuya_cover_options` with `calibration_time_up` and `calibration_time_down`